src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/FindPal.Windows.cs

```diff
@@ -311,11 +337,16 @@ public void FindBySubjectKeyIdentifier(byte[] keyIdentifier)
                 {
                     int cbData = 0;
                     if (!Interop.Crypt32.CertGetCertificateContextProperty(pCertContext, Interop.Crypt32.CertContextPropId.CERT_KEY_IDENTIFIER_PROP_ID, null, ref cbData))
+                    {
                         return false;
+                    }
 
                     byte[] actual = new byte[cbData];
```

Comment:
We could probably do some improvement here.

1) If cbData != keyIdentifier.Length return false without the second dip.
2) If cbData <= some cap, stackalloc.

cbData'll be 20 99% of the time, so even a stackalloc-128 probably eliminates all allocations from this routine.

Reply:
And we could do even better if we did the stackalloc first.

```C#
const int MaxStackAlloc = 64;

if (keyIdentifier.Length <= MaxStackAlloc)
{
    Span<byte> actualKeyId = stackalloc byte[keyIdentifier.Length];

    if (!Interop.Crypt32.CertGetCertificateContextProperty(pCertContext, Interop.Crypt32.CertContextPropId.CERT_KEY_IDENTIFIER_PROP_ID, actualKeyId, ref cbData))
    {
        return false;
    }

    return actualKeyId.Slice(0, cbData).SequenceEquals(keyIdentifier);
}
else
{
    // the current code, more or less.
}
```

Reply:
> If cbData != keyIdentifier.Length return false without the second dip.

I'm a little nervous of that since _some_ CAPI APIs I've seen, on the "give null and I'll give you a length" give you a size that is _at least_ big enough to hold the result, while the second call with the real buffer gives the "real" length.

Agree we can put some things on the stack though.

Reply:
> stackalloc byte[keyIdentifier.Length];

ಠ_ಠ

Reply:
OK,

```C#
Span<byte> actualKeyId = stackalloc byte[64];

if (keyIdentifier.Length <= actualKeyId.Length)
{
   ...
}
```

:smile: